src/index.js

```diff
@@ -3,10 +3,10 @@ import WarmupHandler from "./warmup-handler";
 module.exports = extraCallback => {
     return originalFunction => {
         const warmupHandler = new WarmupHandler(extraCallback);
-        return (event, context, callback) => {
+        return async (event, context, callback) => {
             const warmupCallback = typeof callback === "function" ? callback : context.succeed;
             if (!warmupHandler.checkAndHandleWarmupRequest(event, warmupCallback)) {
-                originalFunction(event, context, callback);
+                await originalFunction(event, context, callback);
```

Comment:
If we change this to `return await originalFunction(event, context, callback)` one can also use the wrapper as:

```js
exports.handler = warmupWrapper(async (event) => {
    return await someAsyncCall();
});
```

Reply:
By the way, I am referring to [Node.js 8.10 runtime support.](https://aws.amazon.com/blogs/compute/node-js-8-10-runtime-now-available-in-aws-lambda/)

Reply:
Done.